### PR TITLE
Refs #37883 - wrap evr remote checks in method

### PR DESCRIPTION
**Pull request by ianballou**
Attempts to fix the issue mentioned here https://github.com/theforeman/foreman-installer/pull/984/files#r1851069746

Wraps all the code in a method and calls the method at the end.

Untested since I'm posting this right at the end of the day in case it can merge early.

### CODE REVIEW

File: hooks/pre_commit/42-evr_extension_permissions.rb [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+    return
+  when '1'
+    fail_and_exit("The evr extension is not owned by the foreman DB owner. Please run the following command to fix it: " \
+                  "UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{config[:username]}') WHERE extname='evr';")
+  else
+    fail_and_exit("Failed to check the ownership of the evr extension.")
+  end
 end
+
+check_remote_evr_extension_permissions
```
Comment (ekohl) [inline hooks/pre_commit/42-evr_extension_permissions.rb:41]: I wonder if we should have guards here
```suggestion
check_remote_evr_extension_permissions if !local_postgresql? && katello_enabled?
```

Comment (ehelms) [reply hooks/pre_commit/42-evr_extension_permissions.rb:41]: The guards are in the method currently -- https://github.com/theforeman/foreman-installer/pull/999/files#diff-690025616ae0636f98279c9e319051813fa73ee28f489541b4dd89b2662756acR3-R4

Comment (ekohl) [reply hooks/pre_commit/42-evr_extension_permissions.rb:41]: I wasn't quite sure what's nicer. It probably doesn't really matter.

What does matter is the trailing whitespace failure that breaks RuboCop.

**Review by ehelms — APPROVED** [review]

**Comment by ianballou** [comment]
Thanks for cleaning  the PR up @ekohl !

**Pull request merged**